src/webapp/components/dataset-wizard/DisaggregationStep.tsx

```diff
@@ -166,10 +196,29 @@ function useGetRelatedDataElements(props: { dataSet: DataSet; settings: DataSetS
                 const allIndicators = Indicator.buildAllIndicators(indicators);
                 setIndicatorsDataElements(allIndicators);
                 setRefresh(refresh + 1);
+                onChange(dataSet.setIndicators(indicators));
             }, console.error);
-    }, [compositionRoot.indicators.getRelated, dataSet, refresh, settings.existingIndicatorIds]);
+    }, [
+        compositionRoot.indicators.getRelated,
```

Comment:
completely minor, but we typically just pass compositionRoot, as the whole object is immutable.